packages/@webex/internal-plugin-voicea/src/voicea.ts

```diff
@@ -341,20 +369,107 @@ export class VoiceaChannel extends WebexPlugin implements IVoiceaChannel {
       body: {
         transcribe: {caption: true},
       },
-    }).then(() => {
-      Trigger.trigger(
-        this,
-        {
-          file: 'voicea',
-          function: 'turnOnCaptions',
-        },
-        EVENT_TRIGGERS.CAPTIONS_TURNED_ON
-      );
-      this.areCaptionsEnabled = true;
-      this.sendAnnouncement();
+    })
+      .then(() => {
+        Trigger.trigger(
+          this,
+          {
+            file: 'voicea',
+            function: 'turnOnCaptions',
+          },
+          EVENT_TRIGGERS.CAPTIONS_TURNED_ON
+        );
+        this.areCaptionsEnabled = true;
+        this.captionStatus = TURN_ON_CAPTION_STATUS.ENABLED;
+        this.announce();
+      })
+      .catch(() => {
+        this.captionStatus = TURN_ON_CAPTION_STATUS.IDLE;
+        throw new Error('turn on captions fail');
+      });
+  };
+
+  /**
+   * bind llm online event once to announce
+   * @returns {void}
+   */
+  private announceAfterLLMOnline = () => {
+    this.announceStatus = ANNOUNCE_STATUS.WAITING_LLM_ONLINE;
+    this.onceLLMOnline(() => {
+      if (this.announceStatus === ANNOUNCE_STATUS.WAITING_LLM_ONLINE) {
+        this.announceStatus = ANNOUNCE_STATUS.IDLE;
+        this.announce();
+      }
+    });
+  };
+
+  /**
+   * is announce processing
+   * @returns {boolean}
+   */
+  private isAnnounceProcessing = () =>
+    [ANNOUNCE_STATUS.JOINING, ANNOUNCE_STATUS.WAITING_LLM_ONLINE, ANNOUNCE_STATUS.JOINED].includes(
+      this.announceStatus
+    );
+
+  /**
+   * announce to voicea data chanel
+   * @returns {void}
+   */
+  public announce = () => {
+    if (this.isAnnounceProcessing()) return;
+    // @ts-ignore
+    if (!this.webex.internal.llm.isConnected()) {
+      this.announceAfterLLMOnline();
+
+      return;
+    }
+    this.sendAnnouncement();
+  };
+
+  /**
+   * bind llm online event once to turn on captions
+   * @returns {void}
+   */
+  private turnOnCaptionsAfterLLMOnline = (): Promise<void> => {
+    this.captionStatus = TURN_ON_CAPTION_STATUS.WAITING_LLM_ONLINE;
+
+    return new Promise((resolve) => {
+      this.onceLLMOnline(() => {
```

Comment:
why dont we move this logic inside the onceLLMonline -> 

```
        if (this.captionStatus === TURN_ON_CAPTION_STATUS.WAITING_LLM_ONLINE) {
          this.captionStatus = TURN_ON_CAPTION_STATUS.IDLE;
          resolve(this.turnOnCaptions());
        }
        ```

Reply:
The problem to be solved here is that when the user clicks to turn on the captions, if the connection of llm is not yet completed, it will fail to turn on. We have added a failure message on the UI, but for guest users, the failure rate of turning on captions for the first time is almost 100%.
So when the user calls the turn-on api, if the llm is not yet connected, it will finish sending the turn-on message when the llm is online. And this message only needs to be sent once, and if the llm reconnects later, it does not need to send the open message again.

Reply:
onceLLMOnline is a utility method that is also used in announce